pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/JSONSchemaHandler.java

```diff
@@ -64,31 +73,23 @@ public Object deserialize(ByteBuf payload) {
         }
 
         payload.readBytes(buffer, 0, size);
-
-        try {
-            return dslJson.deserialize(Map.class, buffer, size);
-        } catch (IOException e) {
-            log.error("Failed to deserialize Json object", e);
-            return null;
-        }
+        return genericJsonSchema.decode(buffer);
     }
 
     @Override
     public Object extractField(int index, Object currentRecord) {
         try {
-            Map jsonObject = (Map) currentRecord;
+            GenericJsonRecord genericJsonRecord = (GenericJsonRecord) currentRecord;
             PulsarColumnHandle pulsarColumnHandle = columnHandles.get(index);
-
-            String[] fieldNames = pulsarColumnHandle.getFieldNames();
-            Object field = jsonObject.get(fieldNames[0]);
-            if (field == null) {
-                return null;
-            }
-            for (int i = 1; i < fieldNames.length ; i++) {
-                field = ((Map) field).get(fieldNames[i]);
-                if (field == null) {
-                    return null;
+            String[] names = pulsarColumnHandle.getName().split("\\.");
```

Comment:
why do we need to split the name by "\\." ?

Reply:
Use ```pulsarColumnHandle.getFieldNames()``` . Splitting the name has performance implications since this is on the critical path

Reply:
I will change this. Thank you for giving me this advice.